test/main/provider/helpers.test.js

```diff
@@ -50,6 +50,24 @@ describe('#getRawTx', () => {
 
     expect(tx.value).toBe('0x0')
   })
+
+  it('should pass through a hex nonce', () => {
+    const tx = getRawTx({ nonce: '0x168' })
+
+    expect(tx.nonce).toBe('0x168')
+  })
+
+  it('should convert a valid integer nonce into hex', () => {
+    const tx = getRawTx({ nonce: '360' })
+
+    expect(tx.nonce).toBe('0x168')
+  })
+
+  it('should convert an invalid integer nonce to undefined', () => {
```

Comment:
do you think we should pass this through or reject? just worried about a case where someone means to set a nonce and accidentally passes bad data, then it still gets broadcast with a value they didn't intend

Reply:
I thought this approach was closer to what we currently have but if we're tightening things up then we should make it so a positive integer string and valid hex string are the only possibilities, and everything else is rejected.

Reply:
I actually think this changes the behavior because we currently don't manipulate the nonce so the transaction would be rejected, but with this change we would auto-fill the nonce when we're about to send to the chain. probably best to just reject invalid nonces

Reply:
if we wanted to keep it as close as possible to current behavior while only supporting valid numeric nonces, we could simply convert only valid positive integers to hex and pass everything else through. this might be the most straightforward way to go